go-controller/pkg/ovn/secondary_layer3_network_controller.go

```diff
@@ -420,8 +420,11 @@ func (oc *SecondaryLayer3NetworkController) newRetryFramework(
 // Start starts the secondary layer3 controller, handles all events and creates all needed logical entities
 func (oc *SecondaryLayer3NetworkController) Start(ctx context.Context) error {
 	klog.Infof("Start secondary %s network controller of network %s", oc.TopologyType(), oc.GetNetworkName())
-
-	if err := oc.Init(ctx); err != nil {
+	_, err := oc.getNetworkID()
```

Comment:
nit: we should probably change this to be setNetworkID() or something, or split the methods into a getter and a setter but doesnt have to be done in this PR.